src/api/rest/v1/connections/controller.ts

```diff
@@ -0,0 +1,302 @@
+ * - User is shown a deeplink which includes connection credentials (access, refresh token)
+ * - User clicks deeplink to open Vault
+ * - Vault fetches connection and profile information from the DATA_PROFILE_SCHEMA datastore
+ * 
+ * Data sync process:
+ * 
+ * - ..
+ */
+export default class Controller {
+
+    /**
+     * Initialize a syncronization for a `Verida: Vault` context
+     * 
+     * @param req 
+     * @param res 
+     * @param next 
+     */
+    public static async sync(req: UniqueRequest, res: Response, next: any) {
+        try {
+            const {
+                forceSync,
+                instantComplete
+            } = req.body
+
+            const networkInstance = await Utils.getNetworkFromRequest(req)
+            const syncManager = new SyncManager(networkInstance.context, req.requestId)
+
+            if (instantComplete) {
+                syncManager.sync(undefined, undefined, forceSync)
+
+                return res.send({
+                    success: true
+                })
+            } else {
+                const connections = await syncManager.sync(undefined, undefined, forceSync)
+
+                return res.send({
+                    connections,
+                    success: true
+                })
+            }
+        } catch (error) {
+            console.log(error)
+            res.status(500).send(error.message);
+        }
+    }
+
+    public static async syncConnection(req: UniqueRequest, res: Response, next: any) {
+        try {
+            const connectionId = req.params.connectionId
+            const {
+                instantComplete,
+                forceSync
+            } = req.body
+
+            const networkInstance = await Utils.getNetworkFromRequest(req)
+            const syncManager = new SyncManager(networkInstance.context, req.requestId)
+            const connection = await syncManager.getConnection(connectionId)
+
+            if (instantComplete) {
+                syncManager.sync(connection.providerId, connection.accountId, forceSync)
+
+                return res.send({
+                    success: true
+                })
+            } else {
+                const connections = await syncManager.sync(connection.providerId, connection.accountId, forceSync)
+
+                return res.send({
+                    connections,
+                    success: true
+                })
+            }
+        } catch (error) {
+            console.log(error)
+            res.status(500).send(error.message);
+        }
+    }
+
+    public static async connections(req: UniqueRequest, res: Response, next: any) {
+        try {
+            const query = req.query
+            const providerName = query.provider ? query.provider.toString() : undefined
+            const providerId = query.providerId ? query.providerId.toString() : undefined
```

Comment:
I think we should expect `providerId` instead of `provider` and `accountId` instead of `providerId`

Reply:
Fixed

Reply:
Note: This is kind of deprecated now as I added support for `/connections/:connectionId`

Reply:
Ah yes, true for the `accountId` we can remove completely. For `providerId` it could be interesting for filtering based on the provider (but don't have the need right now)

Reply:
Actually `GET /connections/:connectionId` is not yet implementedm so `accountId` can stay for now